cmd/internal/loginprovider/tryinorderloginprovider_test.go

```diff
@@ -23,20 +23,27 @@ var _ = gc.Suite(&tryInOrderLoginProviderSuite{})
 func (s *tryInOrderLoginProviderSuite) Test(c *gc.C) {
 	p1 := &mockLoginProvider{err: errors.New("provider 1 error")}
 	p2 := &mockLoginProvider{err: errors.New("provider 2 error")}
-	p3 := &mockLoginProvider{}
+	p3 := &mockLoginProvider{token: "successful-login-token"}
 
 	logger := loggo.GetLogger("juju.cmd.loginprovider")
 	lp := loginprovider.NewTryInOrderLoginProvider(logger, p1, p2)
 	_, err := lp.Login(context.Background(), nil)
 	c.Assert(err, gc.ErrorMatches, "provider 2 error")
 
 	lp = loginprovider.NewTryInOrderLoginProvider(logger, p1, p2, p3)
+	c.Assert(lp.Token(), gc.Equals, "")
```

Comment:
This test is asserting behaviour of a login provider that we haven't laid out in a contract yet. This feeds into the error comment above where we haven't really defined what the behaviours of these providers should be.

Reply:
I hear you, how do we go about fixing that?

Reply:
Hopefully my change to the func signature propose above helps fix this.